tests/epyccel/test_epyccel_sets.py

```diff
@@ -283,3 +283,50 @@ def update_tuple_as_arg():
     pyccel_result = epyccel_update()
     python_result =  update_tuple_as_arg()
     assert python_result == pyccel_result
+
+def test_set_with_list(language):
+    def set_With_list():
+        a = [1.6, 6.3, 7.2]
+        b = set(a)
```

Comment:
With your current code, copies like this won't work if `a` is an argument (unknown length). Is this intentional?

Reply:
Can you give an example, please?

Reply:
```python
def f(a : list[float]):
    b = set(a)
```

Reply:
If I try to translate this function I get:
```

ERROR at annotation (semantic) stage
pyccel:
 |fatal [semantic]: tmp_set_copy.py [1,10]| Cannot handle non-homogenous type index
Pyccel has encountered syntax that has not been implemented yet. Please create an issue at https://github.com/pyccel/pyccel/issues and provide a small example of your problem. Do not forget to specify your target language. (list[float])
```

Reply:
Fixed : 39e344c